addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRDevice.java

```diff
@@ -0,0 +1,481 @@
+
+        switch (portEvent) {
+            case CLOSED:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.DSMR_PORT_CLOSED, portEvent.eventDetails);
+                break;
+            case CONFIGURATION_ERROR:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.CONFIGURATION_ERROR, portEvent.eventDetails);
+                break;
+            case DONT_EXISTS:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.CONFIGURATION_ERROR, portEvent.eventDetails);
+                break;
+            case ERROR:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.ERROR, portEvent.eventDetails);
+                break;
+            case LINE_BROKEN:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.READ_ERROR, portEvent.eventDetails);
+                break;
+            case IN_USE:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.ERROR, portEvent.eventDetails);
+                break;
+            case NOT_COMPATIBLE:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.ERROR, portEvent.eventDetails);
+                break;
+            case OPENED:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.DSMR_PORT_OPENED, portEvent.eventDetails);
+                break;
+            case WRONG_BAUDRATE:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.SWITCH_BAUDRATE, portEvent.eventDetails);
+                break;
+            case READ_ERROR:
+                handleDSMRDeviceEvent(DSMRDeviceEvent.READ_ERROR, portEvent.eventDetails);
+                break;
+            case READ_OK:
+                /* Ignore this event */
+                break;
+            default:
+                logger.warn("Unknown DSMR Port event");
+                handleDSMRDeviceEvent(DSMRDeviceEvent.ERROR, "Unknown port event");
+                break;
+        }
+    }
+
+    /**
+     * Callback for the watchdog
+     *
+     * If the binding and the real device is working correctly this callback return immediately
+     *
+     * Otherwise it will trigger the {@link #handleDeviceState()} state to evaluate the current state
+     */
+    public void alive() {
+        logger.debug("Alive");
+        if (System.currentTimeMillis() - deviceStatus.stateEnteredTS > DSMRDeviceConstants.RECOVERY_TIMEOUT
+                && System.currentTimeMillis()
+                        - deviceStatus.lastTelegramReceivedTS > DSMRDeviceConstants.RECOVERY_TIMEOUT) {
+            logger.debug("Calling handle device state");
+            /*
+             * No update of state of received telegrams for a period of RECOVERY_TIMEOUT
+             * Evaluate current state
+             */
+            executor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    handleDeviceState();
+                }
+            });
+        }
+    }
+
+    @Override
+    public void P1TelegramReceived(List<CosemObject> cosemObjects, String telegramDetails) {
+        if (cosemObjects.size() > 0) {
```

Comment:
Done. Also searched and replaced all other occurences of .size() > 0 with !isEmpty